dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preprocess/DefaultTrackerPreprocessService.java

```diff
@@ -48,15 +48,29 @@
 {
     private List<RuleActionApplier> appliers = new ArrayList<>();
 
+    private List<BundlePreProcessor> preProcessors = new ArrayList<>();
+
     @Autowired( required = false )
     public void setAppliers( List<RuleActionApplier> appliers )
     {
         this.appliers = appliers;
     }
 
+    @Autowired( required = false )
+    public void setPreProcessors( List<BundlePreProcessor> preProcessors )
+    {
+        this.preProcessors = preProcessors;
+    }
+
     @Override
     public TrackerBundle preprocess( TrackerBundle bundle )
     {
+        // TODO we may consider "merging" the BundlePreProcessor with the RuleActionApplier, since
```

Comment:
Just so we dont forget, lets make a jira issue for it (TECH)

Reply:
Done: https://jira.dhis2.org/browse/TECH-477